python/datafusion/dataframe.py

```diff
@@ -175,7 +178,23 @@ def with_column_renamed(self, old_name: str, new_name: str) -> DataFrame:
         Returns:
             DataFrame with the column renamed.
         """
-        return DataFrame(self.df.with_column_renamed(old_name, new_name))
+        return DataFrame(self.df.rename({old_name: new_name}))
+
+    def rename(self, mapping: dict[str, str]) -> DataFrame:
+        r"""Rename one or multiple columns by applying a new projection.
```

Comment:
How about `with_columns_renamed` to be more clear what this is doing? I could imagine users confusing this changing a table's name, even though the dataframe doesn't have a `name` associated with it.

Reply:
I would argue `rename` is quite clear if you are acquainted with pandas and pola-rs API.

With columns renamed is quite verbose ^^

Reply:
I'd like to hear other people's opinions. Also I don't recommend we stray *too far* from what the upstream DataFusion uses for naming conventions.